### PR TITLE
feat(travis) configurable number of repositories to track on each poll

### PR DESCRIPTION
**Pull request by gardleopard**
* add configuration numberOfRepositories for travis
* remove calls to /repos that are not returning results

### CODE REVIEW

File: igor-web/src/main/groovy/com/netflix/spinnaker/igor/travis/service/TravisService.groovy [hunk 1]
```diff
@@ -57,19 +44,21 @@ class TravisService implements BuildService {
...
-        this.travisCache  = travisCache
-        this.artifactRegexes = artifactRegexes == null ? Collections.EMPTY_LIST : new HashSet<>(artifactRegexes)
+    TravisService(String travisHostId, String baseUrl, String githubToken, int numberOfRepositories, TravisClient travisClient, TravisCache travisCache, Iterable<String> artifactRegexes) {
+        this.numberOfRepositories = numberOfRepositories
+        this.groupKey             = "${travisHostId}"
+        this.gitHubAuth           = new GithubAuth(githubToken)
+        this.travisClient         = travisClient
+        this.baseUrl              = baseUrl
+        this.travisCache          = travisCache
+        this.artifactRegexes      = artifactRegexes == null ? Collections.EMPTY_LIST : new HashSet<>(artifactRegexes)
```
Comment (jervi) [inline igor-web/src/main/groovy/com/netflix/spinnaker/igor/travis/service/TravisService.groovy]: Do you want `artifactRegexes` to be a List or a Set? Here ut can be both.

File: igor-web/src/test/groovy/com/netflix/spinnaker/igor/travis/service/TravisServiceSpec.groovy [hunk 2]
```diff
@@ -38,7 +38,7 @@ class TravisServiceSpec extends Specification{
 
     void setup() {
         client = Mock(TravisClient)
-        service = new TravisService('travis-ci', 'http://my.travis.ci', 'someToken', client, null, Collections.emptyList())
+        service = new TravisService('travis-ci', 'http://my.travis.ci', 'someToken', 25, client, null, Collections.emptyList())
```
Comment (jervi) [inline igor-web/src/test/groovy/com/netflix/spinnaker/igor/travis/service/TravisServiceSpec.groovy]: `service = new TravisService('travis-ci', 'http://my.travis.ci', 'someToken', 25, client, null, [])`

Comment (gardleopard) [reply igor-web/src/test/groovy/com/netflix/spinnaker/igor/travis/service/TravisServiceSpec.groovy]: thanks

File: igor-web/src/main/groovy/com/netflix/spinnaker/igor/travis/service/TravisService.groovy (hunk 1 above)
Comment (gardleopard) [reply igor-web/src/main/groovy/com/netflix/spinnaker/igor/travis/service/TravisService.groovy]: thanks

File: igor-web/src/main/groovy/com/netflix/spinnaker/igor/travis/service/TravisService.groovy [hunk 3]
```diff
@@ -211,12 +200,12 @@ class TravisService implements BuildService {
...
                 List<Repo> repos = []
-                getAccounts().accounts.each { account ->
-                    Repos accountRepos = travisClient.repos(getAccessToken(), account.login)
-                    accountRepos.repos.each { repo ->
-                        log.debug "[${account.login}] [${repo.slug}]"
+
+                getAccounts().accounts.findAll({ it.isUser() }).each { account ->
+                    calculatePagination(numberOfRepositories).times { page ->
+                        Repos accountRepos = travisClient.repos(getAccessToken(), account.login, true, TRAVIS_BUILD_RESULT_LIMIT, page * TRAVIS_BUILD_RESULT_LIMIT)
+                        repos.addAll accountRepos.repos
```
Comment (jervi) [inline igor-web/src/main/groovy/com/netflix/spinnaker/igor/travis/service/TravisService.groovy]: `repos << accountRepos.repos`

Comment (gardleopard) [reply igor-web/src/main/groovy/com/netflix/spinnaker/igor/travis/service/TravisService.groovy]: thanks

File: igor-web/src/main/groovy/com/netflix/spinnaker/igor/travis/service/TravisService.groovy [hunk 4]
```diff
@@ -26,22 +26,9 @@ import com.netflix.spinnaker.igor.travis.TravisCache
...
-import com.netflix.spinnaker.igor.travis.client.model.EmptyObject
-import com.netflix.spinnaker.igor.travis.client.model.GithubAuth
-import com.netflix.spinnaker.igor.travis.client.model.Job
-import com.netflix.spinnaker.igor.travis.client.model.Jobs
-import com.netflix.spinnaker.igor.travis.client.model.Repo
-import com.netflix.spinnaker.igor.travis.client.model.RepoRequest
-import com.netflix.spinnaker.igor.travis.client.model.Repos
-import com.netflix.spinnaker.igor.travis.client.model.TriggerResponse
-import com.netflix.spinnaker.igor.travis.client.model.v3.TravisBuildType
+import com.netflix.spinnaker.igor.travis.client.model.*
```
Comment (jervi) [inline igor-web/src/main/groovy/com/netflix/spinnaker/igor/travis/service/TravisService.groovy]: Not a big fan of * imports

Comment (gardleopard) [reply igor-web/src/main/groovy/com/netflix/spinnaker/igor/travis/service/TravisService.groovy]: thank you

File: igor-web/src/main/groovy/com/netflix/spinnaker/igor/travis/service/TravisService.groovy (hunk 3 above)
Comment (jervi) [reply igor-web/src/main/groovy/com/netflix/spinnaker/igor/travis/service/TravisService.groovy]: Or not.

**Pull request merged**